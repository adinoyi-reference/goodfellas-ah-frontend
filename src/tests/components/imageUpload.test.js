import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import ImageUploaderClass, { ImageUploader } from '../../components/articles/imageUpload';
import Root from '../../root';

let wrapped;
let wrapper;

const status = {
  success: true,
  error: false,
};

const imageUploadfn = jest.fn();
const store = mockStore({
  imageUploadReducer: {
    status, uploadedImage: 'url',
  },
});

beforeEach(() => {
  wrapped = shallow(
    <Root>
      <MemoryRouter>
        <ImageUploaderClass
          imageUpload={imageUploadfn}
          store={store}
          status={status}
          uploadedImage="url"
        />
      </MemoryRouter>
    </Root>,
  );

  wrapper = shallow(
    <ImageUploader
      imageUpload={imageUploadfn}
      status={status}
      uploadedImage="url"
    />,
  );
});


describe('Create Articles UI', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('ImageUploader functionality', () => {
  it('should simulate image upload', () => {
    const imageUpload = wrapper.find('input');
    imageUpload.simulate('change', {
      target: {
        files: 'file',
      },
    });
    expect(imageUploadfn).toBeCalledWith('file');
  });
});
