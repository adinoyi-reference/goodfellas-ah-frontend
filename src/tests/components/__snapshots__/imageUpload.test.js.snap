// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Articles UI render features view should render as expected 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Connect(ImageUploader)
      imageUpload={[MockFunction]}
      status={
        Object {
          "error": false,
          "success": true,
        }
      }
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      uploadedImage="url"
    />
  </MemoryRouter>
</Provider>
`;
